src/Microsoft.AspNet.OData.Shared/DataModificationExceptionType.cs

```diff
@@ -0,0 +1,118 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="MessageType.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System;
+
+namespace Org.OData.Core.V1
+{
```

Comment:
Honestly speaking, I'm a bit confused by the use of the work `Type` in `MessageType`, `Exception`, and `DataModificationExceptionType`.
The word `Type` suggests that the class is used for categorization

Reply:
It simply means the type of message we dealing with, the type of exception, the type of data modification exception.